marklogic-client-api/src/test/java/com/marklogic/client/test/rows/ValidateDocTest.java

```diff
@@ -86,10 +92,69 @@ public void xmlSchema() {
         assertEquals(uriCountToWrite, persistedUris.size());
         //expectedUris.forEach(uri -> assertTrue("Did not find URI: " + uri, persistedUris.contains(uri)));
     }
+    @Test
+    public void testXmlSchemaWithStrictMode() {
```

Comment:
It looks like only `testWithJsonSchemaAndInvalidJsons` is verifying that validation is occurring. The other tests are only verifying that "resultRows" is returning rows with expected URIs in them. I don't think this test class needs to care about that. It instead should follow a pattern of:

1. Use fromDocDescriptors (like `testWithJsonSchemaAndInvalidJsons` does) and pass in 2 docs, 1 that's valid and 1 that's invalid (only need 2)
2. Ensure that `write()` is part of the plan
3. Verify that `resultRows()` only returns 1 row
4. Verify that the valid doc is written and the invalid doc is not written

That's all that's needed for each of these tests. We don't need to test different kinds of invalid input, we just need to know that invalid input will be discarded, regardless of why it's invalid.

Reply:
Hi @rjrudin , the "testUsingFromParamAndSchematron" test is also verifying something similar. Out of the the 3 documents sent as attachment only one (doc3.xml) should be returned. (I think currently it is failing due to the xdmp-update permission issue). Do you recommend to add a write operation to it?

Reply:
Yes, I think `write` is good to do for all of these tests to make them more realistic, as I think well over 90% of the time that `validateDoc` is used, it'll be right before docs are written.

Reply:
Hi Rob, left this test unchanged and added a separate test "testWriteWithXmlSchemaWithStrictMode" for testing write.

Reply:
The "lax" test should still have at least one doc that fails. Right now, I can comment out the line for `validateDoc` and the test still passes, meaning that that test isn't really doing anything yet - it's only testing that `fromDocUris` and `joinDoc` work, but we already have separate tests for those. Same is true for the "strict" test here - since all of the docs are valid, we don't know if the validation is actually working or not.